src/main/scala/zio/redis/Command.scala

```diff
@@ -0,0 +1,83 @@
+package zio.redis
+
+import java.time.Instant
+
+import zio.{ Chunk, IO }
+import zio.stream.Stream
+import zio.duration.Duration
+
+import scala.util.matching.Regex
+
+final case class Command[-In, +Out] private[redis] (
+  name: String,
+  input: Command.Input[In],
+  output: Command.Output[Out]
+) {
+
+  // main command interpreter
+  def run(in: In): Out = ???
```

Comment:
How will the dependencies get passed in? In order to run a command, we'll likely need an actual `Redis` backed command interpreter, but all of the commands output `IO[Error, A]` instead of `ZIO[Redis, Error, A]`

Reply:
I'll try to keep the interpretation here, and I'm quite flexible in terms of environment / layer dependencies at this moment. My plan was to get the rough API, get the "feel" of it, and tackle the interpretation in follow-up.

Reply:
Probably `run` will look like this:

```scala
def run(in: In): Queued[RedisClient, RedisError, Out]
```

Actually, it occurs to me instead of `Queued`, we can probably use `ZQuery`, if we're willing to allow `flatMap` at the cost of breaking a pipeline. /cc @adamgfraser

Reply:
Great! I have been delaying a bit on doing this just because there hasn't been other demand outside of Caliban to date and we have had other priorities with the new environment encoding and 1.0 release, but if there is interest here I can extract ASAP so it is something you can depend on.

Reply:
@adamgfraser Nice! Let's sync over discord and work on this as soon as I got this merged and you have some spare cycles.